src/cluster/BifSupport.cc

```diff
@@ -0,0 +1,141 @@
+
+    const auto& eh = zeek::event_registry->Lookup(func->AsFuncPtr()->GetName());
+    if ( ! eh ) {
+        zeek::reporter->Error("event registry lookup of '%s' failed", zeek::obj_desc(func.get()).c_str());
+        return std::nullopt;
+    }
+
+    return cluster::detail::Event(eh, std::move(args), timestamp);
+}
+} // namespace
+
+zeek::ValPtr publish_event(const zeek::ValPtr& topic, zeek::ArgsSpan args) {
+    static const auto& cluster_event_type = zeek::id::find_type<zeek::RecordType>("Cluster::Event");
+    static const auto& broker_event_type = zeek::id::find_type<zeek::RecordType>("Broker::Event");
+
+    if ( args.empty() ) {
+        zeek::emit_builtin_error("no event arguments given");
+        return zeek::val_mgr->False();
+    }
+
+    if ( topic->GetType()->Tag() != zeek::TYPE_STRING ) {
+        zeek::emit_builtin_error("topic is not a string");
+        return zeek::val_mgr->False();
+    }
+
+    const auto topic_str = topic->AsStringVal()->ToStdString();
+
+    auto timestamp = zeek::event_mgr.CurrentEventTime();
+
+    if ( args[0]->GetType()->Tag() == zeek::TYPE_FUNC ) {
+        auto event = zeek::cluster::backend->MakeClusterEvent({zeek::NewRef{}, args[0]->AsFuncVal()}, args.subspan(1),
+                                                              timestamp);
+        if ( event )
+            return zeek::val_mgr->Bool(zeek::cluster::backend->PublishEvent(topic_str, *event));
+
+        return zeek::val_mgr->False();
+    }
+    else if ( args[0]->GetType()->Tag() == zeek::TYPE_RECORD ) {
+        if ( args[0]->GetType() == cluster_event_type ) { // Handling Cluster::Event record type
+            auto ev = to_cluster_event(cast_intrusive<zeek::RecordVal>(args[0]));
+            if ( ! ev )
+                return zeek::val_mgr->False();
+
+            return zeek::val_mgr->Bool(zeek::cluster::backend->PublishEvent(topic_str, *ev));
+        }
+        else if ( args[0]->GetType() == broker_event_type ) {
+            // Handling Broker::Event record type created by Broker::make_event()
+            // only works if the backend is broker_mgr!
+            if ( zeek::cluster::backend != zeek::broker_mgr ) {
+                zeek::emit_builtin_error(
+                    zeek::util::fmt("Publish of Broker::Event record instance with type '%s' to a non-Broker backend",
+                                    zeek::obj_desc_short(args[0]->GetType().get()).c_str()));
+
+                return zeek::val_mgr->False();
+            }
+
+            return zeek::val_mgr->Bool(zeek::broker_mgr->PublishEvent(topic_str, args[0]->AsRecordVal()));
+        }
+        else {
+            zeek::emit_builtin_error(zeek::util::fmt("Publish of unknown record type '%s'",
+                                                     zeek::obj_desc_short(args[0]->GetType().get()).c_str()));
+            return zeek::val_mgr->False();
+        }
+    }
+
+    zeek::emit_builtin_error("publish second argument neither function nor record");
```

Comment:
That might be a bit cryptic but I don't have anything super amazing right now either. :-)